ws-server/src/server.rs

```diff
@@ -275,31 +293,39 @@ async fn background_task(
 	let mut data = Vec::with_capacity(100);
 	let mut method_executors = FutureDriver::default();
 
-	while !stop_server.shutdown_requested() {
+	loop {
 		data.clear();
 
-		if let Err(err) = method_executors.select_with(receiver.receive_data(&mut data)).await {
-			match err {
-				SokettoError::Closed => {
-					tracing::debug!("Remote peer terminated the connection: {}", conn_id);
-					tx.close_channel();
-					return Ok(());
-				}
-				SokettoError::MessageTooLarge { current, maximum } => {
-					tracing::warn!(
-						"WS transport error: message is too big error ({} bytes, max is {})",
-						current,
-						maximum
-					);
-					send_error(Id::Null, &tx, ErrorCode::OversizedRequest.into());
-					continue;
-				}
-				// These errors can not be gracefully handled, so just log them and terminate the connection.
-				err => {
-					tracing::error!("WS transport error: {:?} => terminating connection {}", err, conn_id);
-					tx.close_channel();
-					return Err(err.into());
-				}
+		{
+			// Need the extra scope to drop this pinned future and reclaim access to `data`
+			let receive = receiver.receive_data(&mut data);
+
+			tokio::pin!(receive);
+
+			if let Err(err) = method_executors.select_with(Monitored::new(receive, &stop_server)).await {
+				match err {
+					MonitoredError::Selector(SokettoError::Closed) => {
+						tracing::debug!("Remote peer terminated the connection: {}", conn_id);
```

Comment:
```suggestion
						tracing::debug!("WS transport error: remote peer terminated the connection: {}", conn_id);
```

(not to self, plug in close reason here one day)